homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -105,3 +120,27 @@ def update(self, *_):
     def add_update_listener(self, listener):
         """Add a listener for update notifications."""
         self._update_listeners.append(listener)
+
+    def _register_services(self) -> None:
+        """Register the services for BMW vehicles.
+
+        The different types of services are defined in the _SERVICE_MAP.
+        """
+        for service in _SERVICE_MAP:
+            _LOGGER.debug('Registering service %s', service)
+            self._hass.services.register(
+                DOMAIN, service,
+                self._execute_service,
+                schema=SERVICE_SCHEMA)
+
+    def _execute_service(self, call):
```

Comment:
Make function to handle service calls directly inside `setup` instead. You have access to this instance there, so this doesn't have to be a method here.

Reply:
fixed